pydantic/main.py

```diff
@@ -89,7 +104,7 @@ class BaseConfig:
     getter_dict: Type[GetterDict] = GetterDict
     alias_generator: Optional[Callable[[str], str]] = None
     keep_untouched: Tuple[type, ...] = ()
-    schema_extra: Union[Dict[str, Any], Callable[[Dict[str, Any]], None]] = {}
+    schema_extra: Union[Dict[str, Any], 'SchemaExtraCallable'] = {}
```

Comment:
maybe you could test this works by adding a to `tests/mypy/modules/success.py`?

Reply:
I could use some help here. I've been trying something like this to pass
```py
class ConfigSchemaExtraCallable(BaseConfig['MyModel']):
    @staticmethod
    def schema_extra(schema: Dict[str, Any], model_class: Type['MyModel']) -> None:
        pass
```
to no avail.
Even when I made `BaseConfig` generic, mypy still complained about signature incompatibility.